packages/create-codes/src/helpers/get-current-branch-name.ts

```diff
@@ -0,0 +1,18 @@
+import { execFileSync } from "node:child_process";
+import { resolve } from "node:path";
+
+function executeGitCommand(command: string[]) {
+  return execFileSync("git", [...command], {
+    encoding: "utf-8",
+    cwd: resolve(__dirname, ".."),
+  });
+}
+
+export async function getCurrentBranchName() {
+  const currentBranch = executeGitCommand(["branch", "--show-current"]).trim();
+  // If it's a CI environment, we can simply use the current branch name.
+  if (process.env.CI) return currentBranch;
+
+  const isRemote = executeGitCommand(["ls-remote", "origin", currentBranch]);
+  return isRemote ? currentBranch : "main";
```

Comment:
[Q] Does this mean that if the current branch hasn't been pushed to the remote repository yet, it will return `main`? 
Also, since it's referencing the remote branch, we would still need to push in order to test to most recent changes - is this correct?